### ISSUE TITLE
Entities lost after hassio restart

### ISSUE THREAD

**Issue by jonkristian** [marthoc/hassio-addons#32]
Each time i restart hassio, my deCONZ integration stops working, entities are greyed out.

Steps I take to integrate:

- Go to Configuration > Integrations and click Configure deCONZ Zigbee gateway
- Go to deCONZ UI > Gateway > Advanced > Authenticate
- Go back to Home Assistant UI, click Submit, check both check boxes and then Submit.

Restart hassio, from terminal: `hassio ha restart`

After restart deCONZ integration is still available, but all entities are lost/greyed out. Only way to get back is to re-do the steps above.

deCONZ version: 2.05.44 / 10/28/2018
Firmware: 26280500

**Comment by marthoc** [comment]
At any point can you control your lights? I’m not sure if this is an addon specific problem, maybe @Kane610 can weigh in?

**Comment by Kane610** [comment]
Sounds like it is related to this issue: https://github.com/home-assistant/home-assistant/issues/17082

TLDR; startup takes too long and component times out. See this as a hypothesis since we haven't really ascertained the root cause.

I've added a retry mechanism in the component that will retry if setup fails.

**Comment by jonkristian** [comment]
@nomadzx No sorry have not have time to do that yet, I will during this weekend.

@marthoc Yes, everything is working perfectly once integration is added, however after a hassio restart/reboot nothing works and I have to re-run the integration.

@Kane610 yes, it seems to be similar, however I don't get any errors other than the entity id's are unavailable, and I am already on HassOS 1.11. This retry mechanism, is it available already, or something that will come during next update? If so, hopefully that will close this issue :)

**Comment by Kane610** [comment]
Next release. I merged it to Hass dev yesterday.

**Comment by jonkristian** [comment]
Seems to be stable now :)

**Comment by Kane610** [comment]
Thanks for verifying @jonkristian
